### DISCUSSION TITLE
uvx marimo --sandbox with specific python version

### DISCUSSION THREAD

**Discussion by EtienneT** [marimo-team/marimo#3036] [Q&A]
Love the marimo sandbox mode in combination with uvx, you have everything you need to run your notebooks inside the sandbox.  But I want to edit the marimo notebook with a specific python version.  I have a requirement that can't run in python >3.10.  So I start the notebook like so:

```
uvx --python=3.9.20 marimo edit --sandbox "fda.py"
```

My dependencies are declared like so in my .py:

```
# /// script
# requires-python = "<3.10"
# dependencies = [
#     "matplotlib==3.9.3",
#     "scikit-fda==0.7.0",
# ]
# ///
```

It doesn't seem to respect the python version in either the uvx --python parameter or the requires-python = "<3.10" in the script because when I try to run the note book I get the error:

```
RuntimeError: Cannot install on Python version 3.12.7; only versions >=3.6,<3.10 are supported.

  help: `numba` (v0.53.1) was included because `scikit-fda` (v0.7) depends on `fdasrsf` (v2.6.1) which depends on `numba`
```

I have also tried setting UV_PYTHON=3.9.20 environment variable like mentionned in [marimo documentation here](https://docs.marimo.io/guides/editor_features/package_management.html#choosing-the-python-version), but it doesn't help.

What am I doing wrong?

Thanks,

**Comment by mscolnick** [comment]
You can do `UV_PYTHON=3.13 marimo edit --sandbox notebook.py`


The [docs on this are here](https://docs.marimo.io/guides/editor_features/package_management.html#choosing-the-python-version).

**Reply by EtienneT** [reply]
This does not work in Powershell in windows:

```
UV_PYTHON=3.13: The term 'UV_PYTHON=3.13' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

**Comment by liquidcarbon** [comment]
try `$env:UV_PYTHON=3.13`

or use a newer version? they're at 0.9.1, python>=3.10 https://github.com/GAA-UAM/scikit-fda/blob/develop/pyproject.toml

**Reply by EtienneT** [reply]
Adding a specific version of numba to the list of dependencies in the marimo notebook seems to fix the installation problem.

Thanks for your help guys.
